openwisp_users/multitenancy.py

```diff
@@ -134,3 +136,52 @@ class MultitenantRelatedOrgFilter(MultitenantOrgFilter):
     """
 
     multitenant_lookup = 'organization__in'
+
+
+class BaseOrganizationPermission(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        organization = self.get_object_organization(view, obj)
+        return self.validate_membership(request.user, organization)
+
+    def has_permission(self, request, view):
+        return request.user and request.user.is_authenticated
+
+    def get_object_organization(self, view, obj):
+        organization_field = getattr(view, 'organization_field', 'organization')
```

Comment:
This requirement that is in the original issue is not satisfied:

> In case the organization field is on a parent element, we should be able to do the following:
> 
> ```python
> permission_classes = (isOrganizationMember,)
> organization_field = 'device__organization'
> ```

How can we specify to look on the parent object with the following implementation?